org.eclipse.scanning.api/src/org/eclipse/scanning/api/scan/event/Location.java

```diff
@@ -1,150 +1,150 @@
+ * information.
+ *
+ * @author Matthew Gerring
+ *
+ */
+public class Location implements INameable {
+
+	/**
+	 * Must match method names in IPositionListener as reflection used on the value to
+	 * call the correct method on the listener.
+	 */
+	public enum LocationType {
+		positionWillPerform, levelPerformed, positionChanged, positionPerformed;
+	}
+
+
+	private LocationType type;
+	private String       name;
+	private IPosition    position;
+	private int          level;
+	private List<String> levelNames;
+
+	public Location() {
+
+	}
+	public Location(LocationType t, PositionEvent evnt) {
+		this.type       = t;
+		this.position   = evnt.getPosition();
+		if (position.getNames()!=null&&position.getNames().size()==1) this.name = position.getNames().iterator().next();
+		this.level      = evnt.getLevel();
+		this.levelNames = readLevelNames(evnt.getLevelObjects());
+	}
+
+	private List<String> readLevelNames(List<? extends ILevel> levelObjects) {
+
+		if (levelObjects==null || levelObjects.isEmpty()) return null;
+		List<String> ret = new ArrayList<>();
+		for (ILevel level : levelObjects) {
+			ret.add(level.getName());
+		}
+		return ret;
+	}
+	public LocationType getType() {
+		return type;
+	}
+	public void setType(LocationType type) {
+		this.type = type;
+	}
+	@Override
+	public String getName() {
+		return name;
+	}
+	@Override
+	public void setName(String name) {
+		this.name = name;
+	}
+	public IPosition getPosition() {
+		return position;
+	}
+	public void setPosition(IPosition position) {
+		this.position = position;
+	}
+	public int getLevel() {
+		return level;
+	}
+	public void setLevel(int level) {
+		this.level = level;
+	}
+	public List<String> getLevelNames() {
+		return levelNames;
+	}
+	public void setLevelNames(List<String> levelNames) {
+		this.levelNames = levelNames;
+	}
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + level;
+		result = prime * result + ((levelNames == null) ? 0 : levelNames.hashCode());
+		result = prime * result + ((name == null) ? 0 : name.hashCode());
+		result = prime * result + ((position == null) ? 0 : position.hashCode());
+		result = prime * result + ((type == null) ? 0 : type.hashCode());
+		return result;
+	}
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		Location other = (Location) obj;
+		if (level != other.level)
+			return false;
+		if (levelNames == null) {
+			if (other.levelNames != null)
+				return false;
+		} else if (!levelNames.equals(other.levelNames))
+			return false;
+		if (name == null) {
+			if (other.name != null)
+				return false;
+		} else if (!name.equals(other.name))
+			return false;
+		if (position == null) {
+			if (other.position != null)
+				return false;
+		} else if (!position.equals(other.position))
+			return false;
+		if (type != other.type)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)